src/vcf/fixer.cpp

```diff
@@ -210,10 +243,139 @@ namespace ebi
+                                                       const std::string &separator,
+                                                       const std::string &key_value_separator,
+                                                       const std::string &empty_value)
+        {
+            std::map<std::string, std::string> values;
+            std::vector<std::string> ordered_keys;
+            std::set<std::string> fields_to_remove;
+
+            std::vector<std::string> fields;
+            util::string_split(column, separator.c_str(), fields);
+
+            for (auto & field : fields) {
+                std::vector<std::string> subfields;
+                util::string_split(field, key_value_separator.c_str(), subfields);
+                auto iterator = values.find(subfields[0]);
+                if (iterator == values.end()) {
+                    ordered_keys.push_back(subfields[0]);
+                    values[subfields[0]] = subfields[1];
+                } else if (values[subfields[0]] != subfields[1]) {
+                    fields_to_remove.insert(subfields[0]);
+                }
+            }
+
+            std::string fixed_column;
+            size_t num_removed_duplicates = 0;
+            for (auto & ordered_key : ordered_keys) {
+                if (fields_to_remove.find(ordered_key) == fields_to_remove.end()) {
+                    fixed_column += ordered_key + key_value_separator + values[ordered_key] + separator;
+                } else {
+                    num_removed_duplicates++;
+                }
+            }
+            if (fixed_column.size() == 0) {       // all the fields were removed
+                fixed_column = empty_value;
+            } else {
+                fixed_column.pop_back();          // remove trailing separator (its length is always 1)
+            }
+
+            output << fixed_column;
+            return num_removed_duplicates;
+        }
+
+        size_t Fixer::remove_duplicate_format_sample_pairs(const std::string &string_line) {
+            const size_t format_column_index = 8;
+            size_t num_removed_duplicates = 0;
+
+            using iter = std::vector<std::string>::iterator;
+            fix_columns(format_column_index, -1, string_line, "\t", [&](iter first, iter last) {
+                std::map<std::string, std::vector<size_t>> format_fields;
+                std::vector<std::string> ordered_fields;
+                std::vector<std::string> format_keys;
+                util::string_split(*first, ":", format_keys);
+
+                for (size_t i = 0; i < format_keys.size(); i++) {
+                    format_fields[format_keys[i]].push_back(i);
+                    if (format_fields[format_keys[i]].size() == 1) {
+                        ordered_fields.push_back(format_keys[i]);
+                    }
+                }
+
+                std::set<std::string> fields_to_remove;
+
+                for (auto & format_field : format_fields) {
```

Comment:
this `for` block is building the variable `fields_to_remove`, so it can be extracted into a dedicated function `get_fields_to_remove`